src/Form/Form.js

```diff
@@ -504,33 +513,35 @@ class Form {
         return [...this.inputs.credentials].find((input) => canBeDecorated(input) && isVisible(input))
     }
 
-    promptLoginIfNeeded () {
+    async promptLoginIfNeeded () {
         if (document.visibilityState !== 'visible' || !this.isLogin) return
 
-        if (this.device.settings.availableInputTypes.credentials) {
-            const firstCredentialInput = this.getFirstViableCredentialsInput()
-            const input = this.activeInput || firstCredentialInput
-            if (input) {
-                // The timeout is needed in case the page shows a cookie prompt with a slight delay
-                setTimeout(() => {
-                    // safeExecute checks that the element is on screen according to IntersectionObserver
-                    safeExecute(this.form, () => {
-                        const {x, y, width, height} = this.form.getBoundingClientRect()
-                        const elHCenter = x + (width / 2)
-                        const elVCenter = y + (height / 2)
-                        // This checks that the form is not covered by anything else
-                        const topMostElementFromPoint = document.elementFromPoint(elHCenter, elVCenter)
-                        if (this.form.contains(topMostElementFromPoint)) {
-                            this.execOnInputs((input) => {
-                                if (isVisible(input)) {
-                                    this.touched.add(input)
-                                }
-                            }, 'credentials')
-                            this.device.attachTooltip(this, input, null, 'autoprompt')
-                        }
-                    })
-                }, 200)
-            }
+        const firstCredentialInput = this.getFirstViableCredentialsInput()
+        const input = this.activeInput || firstCredentialInput
+        if (!input) return
+
+        const mainType = getInputMainType(input)
+        const subtype = getInputSubtype(input)
+        if (await this.device.settings.canAutofillType(mainType, subtype)) {
```

Comment:
This is the main change. We now check if the field can be autofilled based on the settings, which is an async call.